File: corpus/csharp/typed_receivers_refusals/.repo/Svc/Dispatch.cs
using System.Collections.Generic;

namespace Svc;

public class Dispatch
{
    private readonly Handler _handler;

    public void ViaField() => _handler.Run();

    public void ViaUnknownVar()
    {
        var h = Make();
        h.Run();
    }

    public void ViaLambda(List<Handler> hs) => hs.ForEach(h => h.Run());

    public void ViaExtension(Handler h) => h.Shout();

    public void ViaShadow()
    {
        Rocket _handler = new Rocket();
        _handler.Run();
    }

    private Handler Make() => new Handler();
}

File: corpus/java/typed_receivers_refusals/.repo/svc/Dispatch.java
package svc;

import java.util.List;

public class Dispatch {
    private final Handler handler = new Handler();

    public void viaField() {
        handler.run();
    }

    public void viaUnknownVar() {
        var r = make();
        r.run();
    }

    public void viaLambdas(List<Handler> hs) {
        hs.forEach(h -> h.run());
        hs.forEach((Handler g) -> g.run());
    }

    public void viaUnionCatch() {
        try {
            make();
        } catch (IllegalStateException | IllegalArgumentException e) {
            e.getMessage();
        }
    }

    public void viaTie(Both b) {
        b.ping();
    }

    public int viaOverride(XProto x) {
        return x.settings();
    }

    public void viaShadow() {
        Rocket handler = new Rocket();
        handler.run();
    }

    public int viaCollision(Lazy z) {
        return z.length();
    }

    private Handler make() {
        return new Handler();
    }
}
